lib/db_connection/ownership/proxy.ex

```diff
@@ -30,40 +31,41 @@ defmodule DBConnection.Ownership.Proxy do
     timeout = Keyword.get(pool_opts, :queue_target, @queue_target) * 2
     interval = Keyword.get(pool_opts, :queue_interval, @queue_interval)
 
+    pre_checkin = Keyword.get(pool_opts, :pre_checkin, fn _, mod, state -> {:ok, mod, state} end)
+    post_checkout = Keyword.get(pool_opts, :post_checkout, &{:ok, &1, &2})
```

Comment:
Do we definitely want to go with funs and not a behavior? As this is for testing only I think funs are preferred but if anyone would use this in prod we should switch to support code reload. I am very doubtful thats the case though. Either way we need to document these keyword values.

Reply:
Yeah, I am still sold on the funs because I find the behaviour too much overhead for something we don't know has a use case without testing. I will add the docs and leave a note.